internal/controllers/core/podlogstream/podlogstreamcontroller.go

```diff
@@ -433,9 +435,16 @@ func (r *Controller) updateStatus(streamName types.NamespacedName) {
 		return
 	}
 
+	lastUpdate, hasLastUpdate := r.lastUpdate[streamName]
+	if hasLastUpdate && cmp.Equal(status, lastUpdate) {
+		return
+	}
+
+	r.lastUpdate[streamName] = status.DeepCopy()
```

Comment:
Shouldn't this only be updated if the status update actually succeeds below?

Reply:
done